grove/blackbox_algorithms/bernstein_vazirani.py

```diff
@@ -0,0 +1,71 @@
+"""Module for the Bernstein-Vazirani Algorithm."""
+
+import pyquil.quil as pq
+from pyquil.gates import *
+import numpy as np
+import utils as bbu
+from blackbox import AbstractBlackBoxAlgorithm
+
+class BernsteinVaziraniAlgorithm(AbstractBlackBoxAlgorithm):
+    def __init__(self, n, vec_a, b):
+        """
+        Creates an instance of an object to simulate the Bernstein-Vazirani Algorithm
+        such that the function f(x) is given by a * x + b, where a is a bitwise dot product and all additions are mod 2.
+        :param n: the number of bits in the domain of the function
+        :param vec_a: a length n vector of 0s and 1s
+        :param b: a binary bit, either 0 or 1
+        """
```

Comment:
missing `:return:` and `:rtype:`, among other things.

Please reference, for example, [here](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html) for reST docstrings, and please apply to the rest of the docstrings.